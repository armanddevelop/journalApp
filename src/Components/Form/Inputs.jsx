import PropTypes from "prop-types";
export const Inputs = ({ page }) => {
  return (
    <>
      {page === "login" ? (
        <>
          <input
            className="auth__input"
            type="text"
            placeholder="Email"
            name="email"
            autoComplete="off"
          />
          <input
            className="auth__input"
            type="password"
            placeholder="password"
            name="password"
          />
        </>
      ) : (
        <>
          <input
            className="auth__input"
            type="text"
            placeholder="Name"
            name="name"
            autoComplete="off"
          />
          <input
            className="auth__input"
            type="text"
            placeholder="Email"
            name="email"
            autoComplete="off"
          />
          <input
            className="auth__input"
            type="password"
            placeholder="password"
            name="password"
          />
          <input
            className="auth__input"
            type="password"
            placeholder="Confirm password"
            name="confirmPassword"
          />
        </>
      )}
    </>
  );
};
Inputs.propTypes = {
  page: PropTypes.string.isRequired,
};
