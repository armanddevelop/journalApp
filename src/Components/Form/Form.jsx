import { Inputs } from "./Inputs";
import PropTypes from "prop-types";
export const Form = ({ pageName, page }) => {
  return (
    <form>
      <Inputs page={page} />
      <button className="btn btn-primary btn-block mb-5" type="submit">
        {pageName}
      </button>
    </form>
  );
};

Form.propTypes = {
  pageName: PropTypes.string.isRequired,
  page: PropTypes.string.isRequired,
};
