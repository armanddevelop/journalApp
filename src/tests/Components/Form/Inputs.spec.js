import { mount } from "enzyme";
import { Inputs } from "../../../Components/Form/Inputs";

describe("Test in Inputs component", () => {
  const page = "login";
  const values = {
    email: "prueba@test.com",
    password: "123466",
    name: "alicha",
    confirmPassword: "123466",
  };
  const handleInputChange = jest.fn();
  const wrapper = mount(
    <Inputs page={page} values={values} handleInputChange={handleInputChange} />
  );

  it("Snapshot in Inputs component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should only show 2 inputs by mount login inputs", () => {
    const inputs = wrapper.find("input").length;

    expect(inputs).toBe(2);
  });

  it("Should have the input email and password the same vale as prop values", () => {
    const emailField = wrapper.find("input[name='email']").props();
    const passwordField = wrapper.find("input[name='password']").props();

    expect(passwordField.value).toEqual(values.password);
    expect(emailField.value).toEqual(values.email);
  });

  it("Should not exist name field and confirmPassword field", () => {
    const nameField = wrapper.find("input[name='name']");
    const confirmPassword = wrapper.find("input[name='confirmPassword']");

    expect(nameField.exists()).toEqual(false);
    expect(confirmPassword.exists()).toEqual(false);
  });

  it("Should fire the onChange function in email field", () => {
    const emailField = wrapper.find("input[name='email']");

    emailField.simulate("change");

    expect(handleInputChange).toHaveBeenCalled();
  });

  it("Should fire the onChange function in password field", () => {
    const emailField = wrapper.find("input[name='password']");

    emailField.simulate("change");

    expect(handleInputChange).toHaveBeenCalled();
  });

  it("SnapShot with the register field form", () => {
    const wrapper = mount(
      <Inputs
        page="register"
        values={values}
        handleInputChange={handleInputChange}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should exist name, email, password, confirm password field", () => {
    const wrapper = mount(
      <Inputs
        page="register"
        values={values}
        handleInputChange={handleInputChange}
      />
    );
    const nameField = wrapper.find("input[name='name']");
    const emailField = wrapper.find("input[name='email']");
    const passwordField = wrapper.find("input[name='password']");
    const confirmPasswordField = wrapper.find("input[name='confirmPassword']");

    expect(nameField.exists()).toBe(true);
    expect(emailField.exists()).toBe(true);
    expect(passwordField.exists()).toBe(true);
    expect(confirmPasswordField.exists()).toBe(true);
  });
});
