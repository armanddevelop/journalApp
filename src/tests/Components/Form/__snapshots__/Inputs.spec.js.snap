// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Inputs component SnapShot with the register field form 1`] = `
Array [
  <input
    autoComplete="off"
    className="auth__input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value="alicha"
  />,
  <input
    autoComplete="off"
    className="auth__input"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value="prueba@test.com"
  />,
  <input
    className="auth__input"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value="123466"
  />,
  <input
    className="auth__input"
    name="confirmPassword"
    onChange={[Function]}
    placeholder="Confirm password"
    type="password"
    value="123466"
  />,
]
`;

exports[`Test in Inputs component Snapshot in Inputs component 1`] = `
Array [
  <input
    autoComplete="off"
    className="auth__input"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value="prueba@test.com"
  />,
  <input
    className="auth__input"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value="123466"
  />,
]
`;
